internal/pkg/agent/cmd/inspect.go

```diff
@@ -5,383 +5,444 @@
 package cmd
 
 import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
 	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v2"
+
+	"github.com/elastic/elastic-agent-libs/logp"
+	"github.com/elastic/elastic-agent-libs/service"
 
-	"github.com/elastic/elastic-agent/internal/pkg/agent/configuration"
+	"github.com/elastic/elastic-agent/internal/pkg/agent/application/paths"
+	"github.com/elastic/elastic-agent/internal/pkg/agent/errors"
+	"github.com/elastic/elastic-agent/internal/pkg/agent/transpiler"
+	"github.com/elastic/elastic-agent/internal/pkg/capabilities"
 	"github.com/elastic/elastic-agent/internal/pkg/cli"
+	"github.com/elastic/elastic-agent/internal/pkg/composable"
 	"github.com/elastic/elastic-agent/internal/pkg/config"
+	"github.com/elastic/elastic-agent/internal/pkg/config/operations"
+	"github.com/elastic/elastic-agent/pkg/component"
+	"github.com/elastic/elastic-agent/pkg/core/logger"
 )
 
 func newInspectCommandWithArgs(s []string, streams *cli.IOStreams) *cobra.Command {
 	cmd := &cobra.Command{
 		Use:   "inspect",
 		Short: "Shows configuration of the agent",
-		Long:  "Shows current configuration of the agent",
-		Args:  cobra.ExactArgs(0),
+		Long: `Shows current configuration of the agent.
+
+By default variable substitution is not performed. Use the --variables flag to enable variable substitution. The
+first set of computed variables are used when only the --variables flag is defined. This can prevent some of the
+dynamic provides (kubernetes, docker, etc.) from providing all the possible variables it could have discovered if given
+more time. The --variables-timeout allows an amount of time to be provided for variable discovery, when set it will
+wait that amount of time before using the variables for the configuration.
+`,
```

Comment:
@blakerouse I am not sure if `timeout` is the right term here for the action? When I read it I though it was a deadline before the substitution would fail? It more like `wait this amount of time` before returning this substitution? Would be great to have @dedemorton's input for this.

Reply:
I changed it to `--variables-wait`. Being that most of the descriptions talk about waiting, the name matches the description. It is also similar to `kubectl wait` that waits for a condition to occur so I think `--variables-wait` would be a better name.

Reply:
Yes, I think it's a better wording thanks for making the change.